app/src/main/java/com/sdp/movemeet/utility/LocationFetcher.java

```diff
@@ -7,40 +7,39 @@
 import android.widget.Toast;
 
 import androidx.core.app.ActivityCompat;
-import androidx.fragment.app.Fragment;
 
 import com.google.android.gms.location.FusedLocationProviderClient;
 import com.google.android.gms.maps.OnMapReadyCallback;
 import com.google.android.gms.maps.SupportMapFragment;
 import com.google.android.gms.tasks.OnSuccessListener;
 import com.google.android.gms.tasks.Task;
 
-public class LocationFetcher {
+public abstract class LocationFetcher {
```

Comment:
It seems like this class's functionality relies on asynchronous operations but the results we want can be accessed synchronously. I think this has caused issues in tests when retrieving the location for instance, as the class returns null since the operation has not yet completed. We should probably refactor this class such that we can wait for the results to succeed or fail before being able to use them.

Reply:
In the previous version of my code, the currentLocation was not initialized before calling the LocationFetcher, hence it could have a null value indeed. I've added a "defaultLocation" that is always assigned to the currentLocation on creation of the fragment to prevent that from happening.